src/core/net/dnssd_server.hpp

```diff
@@ -188,19 +127,176 @@ class Server : public InstanceLocator, private NonCopyable
             }
         }
 
-    private:
         static bool MatchCompressedName(const Message &aMessage, uint16_t aOffset, const char *aName)
         {
             return aOffset != kUnknownOffset && Name::CompareName(aMessage, aOffset, aName) == kErrorNone;
         }
 
+    private:
         const char *mDomainName;         // The serialized domain name.
         uint16_t    mDomainNameOffset;   // Offset of domain name serialization into the response message.
         uint16_t    mServiceNameOffset;  // Offset of service name serialization into the response message.
         uint16_t    mInstanceNameOffset; // Offset of instance name serialization into the response message.
         uint16_t    mHostNameOffset;     // Offset of host name serialization into the response message.
     };
 
+    /**
+     * This enumeration specifies a dns-sd query type.
+     *
+     */
+    enum DnsQueryType : uint8_t
+    {
+        kDnsQueryNone        = OT_DNSSD_QUERY_TYPE_NONE,    ///< Service type unspecified.
+        kDnsQueryBrowse      = OT_DNSSD_QUERY_TYPE_BROWSE,  ///< Service type browse service.
+        kDnsQueryResolve     = OT_DNSSD_QUERY_TYPE_RESOLVE, ///< Service type resolve service instance.
+        kDnsQueryResolveHost = OT_DNSSD_QUERY_TYPE_HOST,    ///< Service type resolve hostname.
+    };
+
+public:
+    /**
+     * This class contains the compress information for a dns packet.
+     *
+     */
+    class QueryTransaction
```

Comment:
Keep `QueryTransaction` private?

Reply:
Make `QueryTransaction` private is not possible since we need to call `Server::GetNextQuery(const QueryTransaction *)` outside the class anyway. We can use `otDnssdQuery *` type which is in fact `void *` but I believe it's not in line with the OpenThread style.
Another point for making this public is that users can ony get const reference to the queries. Accessing the memories will be harmless.

Reply:
> Making `QueryTransaction` private is not possible since we need to call `Server::GetNextQuery(const QueryTransaction *)` outside the class anyway. We can use `otDnssdQuery *` type which is in fact `void *` but I believe it's not in line with the OpenThread style.

I think it's fine to use `otDnssdQuery *` in Server.
Making `QueryTransaction` minimizes the code change, which can be much smaller than the current change?